Modules/Loadable/SubjectHierarchy/Testing/Python/SubjectHierarchyGenericSelfTest.py

```diff
@@ -428,9 +410,135 @@ def section_TestCircularParenthood(self):
     sceneItemID = shNode.GetSceneItemID()
     mainfolder_ID = shNode.CreateFolderItem(sceneItemID, "Main Folder")
     subfolder_ID = shNode.CreateFolderItem(sceneItemID, "Sub Folder")
-    shNode.SetItemParent(subfolder_ID, mainfolder_ID) # Regular hiearchy setting
+    shNode.SetItemParent(subfolder_ID, mainfolder_ID) # Regular hierarchy setting
     shNode.SetItemParent(mainfolder_ID, subfolder_ID) # Makes slicer crash instead of returning an error
 
+  # ------------------------------------------------------------------------------
+  def section_AttributeFilters(self):
+    self.delayDisplay("Attribute filters",self.delayMs)
+
+    import SampleData
+    sceneFile = SampleData.downloadFromURL(
+      fileNames=self.attributeFilterTestSceneFileName,
+      uris=self.attributeFilterTestSceneFileUrl,
+      # loadFiles=True,
+      checksums=self.attributeFilterTestSceneChecksum)[0]
+    if not os.path.exists(sceneFile):
+      logging.error('Failed to download attribute filter test scene to path ' + str(sceneFile))
+    self.assertTrue(os.path.exists(sceneFile))
+
+    slicer.mrmlScene.Clear()
+    ioManager = slicer.app.ioManager()
+    ioManager.loadFile(sceneFile)
+
+    shNode = slicer.mrmlScene.GetSubjectHierarchyNode()
+
+    # Check scene validity
+    self.assertEqual(114, slicer.mrmlScene.GetNumberOfNodes())
+    self.assertEqual(9, shNode.GetNumberOfItems())
+    self.assertEqual(slicer.mrmlScene.GetNumberOfNodesByClass('vtkMRMLMarkupsNode'), 7)
+
+    # Create test SH tree view
+    shTreeView = slicer.qMRMLSubjectHierarchyTreeView()
+    shTreeView.setMRMLScene(slicer.mrmlScene)
+    shTreeView.show()
+
+    shProxyModel = shTreeView.sortFilterProxyModel()
+
+    def testAttributeFilters(filteredObject, proxyModel):
+      # Check include node attribute name filter
+      self.assertEqual(shProxyModel.acceptedItemCount(shNode.GetSceneItemID()), 9)
+      filteredObject.includeNodeAttributeNamesFilter = ['Markups.MovingInSliceView']
+      self.assertEqual(shProxyModel.acceptedItemCount(shNode.GetSceneItemID()), 6)
+      filteredObject.addNodeAttributeFilter('Sajt')
```

Comment:
It is funny, but maybe it is better to change it to something that conveys the intent better for English-speaking readers ('dummy' or 'nonexisting'), or add a comment to the line that it is just some random string (corresponding to a non-existing  attribute name)

Reply:
I realized this as well. I created a test scene first to test the filters, but then the test snippet became the automatic test code and the scene the test scene, and now it would be a hassle replacing the file. If this is not too big a problem I would keep it as is.

Reply:
OK, just add a comment why "Sajt" (is there an attribute like this in the scene?)

Reply:
I added an ASCII tree as comment in this function